src/components/utils/IntegrationSwitcher.tsx

```diff
@@ -0,0 +1,47 @@
+import { Button, Divider, Drawer, List, ListItem, ListItemButton, Typography } from "@mui/material";
+import { useRouter } from "next/router";
+import { useState } from "react";
+
+import IntegrationLogo from "@/components/utils/IntegrationLogo";
+import activeIntegrations from "@/lib/integrations/active";
+import { IntegrationProfile } from "@/types/integration";
+
+function SelectIntegration({ currentIntegrationProfile }: { currentIntegrationProfile: IntegrationProfile }) {
+    const [open, setOpen] = useState(false);
+    const router = useRouter();
+    return (
+        <>
+            <Button onClick={() => setOpen(true)}>
+                <IntegrationLogo integrationProfile={currentIntegrationProfile} />
+            </Button>
+            <Drawer anchor={"left"} open={open} onClose={() => setOpen(false)}>
+                <List>
+                    <ListItem>
+                        <Typography variant={"h6"}>Velg ditt treningssenter</Typography>
+                    </ListItem>
+                    <Divider />
+                    {Object.values(activeIntegrations).map((integration) => (
+                        <ListItem
+                            value={integration.profile.acronym}
+                            key={integration.profile.acronym}
+                            onClick={async () => {
+                                if (router.asPath.includes(integration.profile.acronym)) {
+                                    return setOpen(false);
+                                }
+                                await router.push(`/${integration.profile.acronym}`);
+                                // Some components do not rerender when initialSchedule is updated
+                                router.reload();
```

Comment:
Would be cool to not have to reload the page here, since it introduces a slight delay between the user clicking the navigation, and the page being ready. If you have any ideas here, let me know. 🤔